src/aelin-pool.ts

```diff
@@ -145,9 +146,21 @@ export function handlePurchasePoolToken(event: PurchasePoolTokenEvent): void {
   poolCreatedEntity.contributions = poolCreatedEntity.contributions.plus(
     event.params.purchaseTokenAmount
   );
-  poolCreatedEntity.save();
 
+  let depositEntity = new Deposit(

```

Comment:
we can probably get rid of the PurchasePoolToken entity in the near future but this is fine for now